libs/ml/tests/unit/graph/graph.cpp

```diff
@@ -397,24 +397,24 @@ TYPED_TEST(GraphTest, variable_freezing_subgraph)
   auto gradient_vector_2 = g.GetGradients();
   for (auto grad : gradient_vector_2)
   {
-    grad.Fill(static_cast<DataType>(2.0));
+    grad.Fill(fetch::math::Type<DataType>("2.0"));
   }
   g.ApplyGradients(gradient_vector);
 
   auto weights_3 = g.GetWeights();
 
-  ASSERT_FALSE(weights_2.at(0).AllClose(weights_3.at(0), static_cast<DataType>(0),
-                                        static_cast<DataType>(0)));
-  ASSERT_FALSE(weights_2.at(1).AllClose(weights_3.at(1), static_cast<DataType>(0),
-                                        static_cast<DataType>(0)));
-  ASSERT_FALSE(weights_2.at(2).AllClose(weights_3.at(2), static_cast<DataType>(0),
-                                        static_cast<DataType>(0)));
-  ASSERT_FALSE(weights_2.at(3).AllClose(weights_3.at(3), static_cast<DataType>(0),
-                                        static_cast<DataType>(0)));
-  ASSERT_FALSE(weights_2.at(4).AllClose(weights_3.at(4), static_cast<DataType>(0),
-                                        static_cast<DataType>(0)));
-  ASSERT_FALSE(weights_2.at(5).AllClose(weights_3.at(5), static_cast<DataType>(0),
-                                        static_cast<DataType>(0)));
+  ASSERT_FALSE(weights_2.at(0).AllClose(weights_3.at(0), fetch::math::Type<DataType>("0"),
```

Comment:
(optional) DataType{0}

Reply:
I would say that's not optional; static_cast<DataType>(0) is better than the proposed change; DataType{0} is also fine